### ISSUE TITLE
Stackdriver - aggregating series returns more than one series

### ISSUE THREAD

**Issue by amit-o** [grafana/grafana#13914]
Please include this information:

### What Grafana version are you using?
5.3.2

### What datasource are you using?
Stackdriver

### What OS are you running grafana on?
Official docker image

### What did you do?
Deployed a container with 2 replicas or had a container that was evicted from a node
Created a graph using the default metric container.googleapis.com - CPU usage, grouped by resource.label.container_name

### What was the expected result?
One series for the same container name

### What happened instead?
Graph had multiple lines as the response from google contained more than one series, despite having the same container_name.
I assume the fact that I'm unable to choose a different aggregation/aligner is also a part of this issue

### Include raw network request & response: get by opening Chrome Dev Tools (F12, Ctrl+Shift+I on windows, Cmd+Opt+I on Mac), go the network tab.

```json
{"from":"1540981583736","to":"1540985183736","queries":[{"refId":"A","intervalMs":5000,"datasourceId":1,"metricType":"container.googleapis.com/container/cpu/usage_time","primaryAggregation":"","perSeriesAligner":"ALIGN_DELTA","alignmentPeriod":"stackdriver-auto","groupBys":["resource.label.container_name"],"view":"FULL","filters":["resource.label.cluster_name","=","stg-k8s","AND","resource.label.container_name","=","company-api"],"aliasBy":"{{resource.label.container_name}}","type":"timeSeriesQuery"}]}
```

```json
{"results":{"A":{"refId":"A","meta":{"alignmentPeriod":60,"groupBys":["resource.label.container_name"],"metricLabels":{},"rawQuery":"aggregation.alignmentPeriod=%2B60s\u0026aggregation.crossSeriesReducer=REDUCE_NONE\u0026aggregation.groupByFields=resource.label.container_name\u0026aggregation.perSeriesAligner=ALIGN_DELTA\u0026filter=metric.type%3D%22container.googleapis.com%2Fcontainer%2Fcpu%2Fusage_time%22+resource.label.cluster_name%3D%22stg-k8s%22+resource.label.container_name%3D%22test-container%22\u0026interval.endTime=2018-10-31T11%3A27%3A15Z\u0026interval.startTime=2018-10-31T10%3A27%3A15Z\u0026view=FULL","resourceLabels":{"cluster_name":["stg-k8s"],"container_name":["company-api"],"instance_id":["1158918389286569730","2704882598942345032","7555989168398534562"],"namespace_id":["89b77d0b-851c-11e7-8019-42010a84024b"],"pod_id":["1620b522-d7a5-11e8-b1ea-42010a8400c3","161eeb51-d7a5-11e8-b1ea-42010a8400c3","1620ea3a-d7a5-11e8-b1ea-42010a8400c3"],"project_id":["..."],"zone":["europe-west1-c"]},"resourceTypes":["gke_container"]},"series":[{"name":"company-api","points":[[0.01793370899997626,1540981695000],[0.01728359700001647,1540981755000],[0.017354004875016926,1540981815000],[0.017565228500018293,1540981875000],[0.016578871874969536,1540981935000],[0.013619801999993797,1540981995000],[0.017326173750007,1540982055000],[0.02844528899998977,1540982115000],[0.024959769125018738,1540982175000],[0.014503209499991954,1540982235000],[0.015509083375036425,1540982295000],[0.01852670499999931,1540982355000],[0.01822547849997136,1540982415000],[0.017321799000001192,1540982475000],[0.018090050875002817,1540982535000],[0.02039480650000769,1540982595000],[0.01928300824999951,1540982655000],[0.015947613499974977,1540982715000],[0.016675516999953288,1540982775000],[0.018859227500001907,1540982835000],[0.01919207962504288,1540982895000],[0.020190635999995266,1540982955000],[0.019266266750037175,1540983015000],[0.016493158999992374,1540983075000],[0.017569766750000326,1540983135000],[0.02079958999996734,1540983195000],[0.019926623749995542,1540983255000],[0.01730772500002331,1540983315000],[0.017308740499970554,1540983375000],[0.017311787000039658,1540983435000],[0.016908800375006194,1540983495000],[0.015699840499962647,1540983555000],[0.016008858125019287,1540983615000],[0.016935910999961834,1540983675000],[0.018275490375003756,1540983735000],[0.022294228500015834,1540983795000],[0.021309894375008298,1540983855000],[0.01835689199998569,1540983915000],[0.018135726375021477,1540983975000],[0.017472229500015146,1540984035000],[0.0171456854999974,1540984095000],[0.016166053500001,1540984155000],[0.016155611999977282,1540984215000],[0.016124287500019818,1540984275000],[0.017848538375005774,1540984335000],[0.02302129099996364,1540984395000],[0.02089352149999968,1540984455000],[0.014510213000050953,1540984515000],[0.015269593374966917,1540984575000],[0.017547734499999024,1540984635000],[0.01689793487503266,1540984695000],[0.014948535999963042,1540984755000],[0.015788941250036714,1540984815000],[0.018310156999973515,1540984875000],[0.01744364537501042,1540984935000],[0.014844110500007446,1540984995000],[0.01543964587500568,1540985055000],[0.017226252000000386,1540985115000],[0.01291968900000029,1540985175000]]},{"name":"company-api","points":[[0.016749003124971296,1540981695000],[0.017740416500032552,1540981755000],[0.01741524074998324,1540981815000],[0.016439713500005837,1540981875000],[0.016796454000029826,1540981935000],[0.017866675499988105,1540981995000],[0.017791951999981848,1540982055000],[0.01756778150001992,1540982115000],[0.017210129750026226,1540982175000],[0.016137174499988305,1540982235000],[0.016404731249963334,1540982295000],[0.017207401500002106,1540982355000],[0.017451177624991487,1540982415000],[0.018182506000016474,1540982475000],[0.01813119562501697,1540982535000],[0.017977264500018464,1540982595000],[0.017930470875000992,1540982655000],[0.01779009000000542,1540982715000],[0.0181957817499665,1540982775000],[0.019412857000020267,1540982835000],[0.020979155249960968,1540982895000],[0.025678050000010444,1540982955000],[0.023089449499991588,1540983015000],[0.015323647999991863,1540983075000],[0.015710727125053836,1540983135000],[0.01687196449995554,1540983195000],[0.017221325249977326,1540983255000],[0.01826940750004269,1540983315000],[0.01796929562499372,1540983375000],[0.01706896000001734,1540983435000],[0.017291453374980392,1540983495000],[0.017958933499983232,1540983555000],[0.018179349875026674,1540983615000],[0.01884059899998647,1540983675000],[0.018520022625011734,1540983735000],[0.017558293499973843,1540983795000],[0.017602615875034644,1540983855000],[0.017735582999989674,1540983915000],[0.017031618625026113,1540983975000],[0.014919725499964898,1540984035000],[0.016166261625016887,1540984095000],[0.019905870000002324,1540984155000],[0.018688911625019955,1540984215000],[0.015038036499959162,1540984275000],[0.01702544099998704,1540984335000],[0.022987654500013832,1540984395000],[0.021572413000001234,1540984455000],[0.017326688500020282,1540984515000],[0.017297785249979825,1540984575000],[0.017211075500028983,1540984635000],[0.017289917124969634,1540984695000],[0.01752644200001896,1540984755000],[0.04803143149996458,1540984815000],[0.13954640000002883,1540984875000],[0.11011337212499939,1540984935000],[0.021814288499967915,1540984995000],[0.021171414750028816,1540985055000],[0.019242793499984145,1540985115000],[0.01443209512501653,1540985175000]]},{"name":"company-api","points":[[0.027470649000008507,1540981695000],[0.024897872999986248,1540981755000],[0.023748281374992075,1540981815000],[0.020299506500009556,1540981875000],[0.020039643749953484,1540981935000],[0.01926005550001264,1540981995000],[0.019224189250053314,1540982055000],[0.019116590500004804,1540982115000],[0.019321461375000126,1540982175000],[0.019936073999986093,1540982235000],[0.020057209249955577,1540982295000],[0.02042061500003456,1540982355000],[0.019593313499967735,1540982415000],[0.017111408999994637,1540982475000],[0.017686609000008957,1540982535000],[0.019412208999995073,1540982595000],[0.021000992375036276,1540982655000],[0.025767342499989354,1540982715000],[0.02472162662502342,1540982775000],[0.02158447899995508,1540982835000],[0.021218562624994775,1540982895000],[0.020120813500000168,1540982955000],[0.01967356950001431,1540983015000],[0.018331837499999892,1540983075000],[0.0187580292500229,1540983135000],[0.02003660449997824,1540983195000],[0.020266710875034732,1540983255000],[0.02095702999997684,1540983315000],[0.021662859125001432,1540983375000],[0.023780346500018368,1540983435000],[0.024511986999982582,1540983495000],[0.026706908499988913,1540983555000],[0.024890259125015746,1540983615000],[0.019440310999982557,1540983675000],[0.019691807250012516,1540983735000],[0.020446295999988706,1540983795000],[0.020077219625022735,1540983855000],[0.018969990500011136,1540983915000],[0.019291305874958198,1540983975000],[0.020255252000026758,1540984035000],[0.02016690524999376,1540984095000],[0.019901865000008456,1540984155000],[0.021692137749994345,1540984215000],[0.027062956000008853,1540984275000],[0.02510108412502632,1540984335000],[0.019215468499965027,1540984395000],[0.0195648788749736,1540984455000],[0.020613109999999324,1540984515000],[0.020312034375024268,1540984575000],[0.019408807499985414,1540984635000],[0.019427711249988988,1540984695000],[0.01948442249999971,1540984755000],[0.019014498999979423,1540984815000],[0.017604728500032252,1540984875000],[0.020216718000028777,1540984935000],[0.028052686500018353,1540984995000],[0.025628172249980707,1540985055000],[0.018354629499981456,1540985115000],[0.013765972124986092,1540985175000]]}],"tables":null}}}
```

**Comment by torkelo** [comment]
not sure if this is a Grafana or Google Stackdriver issue

**Comment by sunker** [comment]
This might actually be the same issue that was recently fixed in #13784. 
@amit-o, please have a look at [this ](https://github.com/grafana/grafana/issues/13691#issuecomment-431354284) picture. Do you see more than one Resource Type in the Metrics Explorer in Stackdriver? In that case you're getting one graph for each resource type. #13784 let's you filter and aggregate on resource type. The fix will be in 5.3.2.

**Comment by amit-o** [comment]
@sunker Apologies for the late reply.
I'm using the default GKE metics, there is only one resource type under it
![screen shot 2018-11-04 at 14 56 31](https://user-images.githubusercontent.com/21155198/47964380-e38d8980-e041-11e8-98ef-b92e06554877.png)

**Comment by amit-o** [comment]
@torkelo I believe this is an issue with Grafana, as it's replicated in metrics that do not have an aggregate selectable in the ui
![screen shot 2018-11-04 at 15 11 35](https://user-images.githubusercontent.com/21155198/47964526-15075480-e044-11e8-9604-83825f8e5da1.png)

**Comment by amit-o** [comment]
That's incredibly weird.
I created a new cluster version 1.11.2-gke.18
I filtered by resource.label.container_name = dnsmasq and by cluster name and saw 2 timeseries (dnsmasq is deployed by default with 2 replicas)
I'm on Grafana v5.3.2 (0d821d0), perhaps it is somehow fixed in a different version?
![screen shot 2018-11-20 at 15 42 42](https://user-images.githubusercontent.com/21155198/48777348-058e3980-ecdb-11e8-8446-26299427c6d3.png)

**Comment by bsmarcosj** [comment]
I think I've a similar problem but when the alert is checked instead of in the graph.

I have two instances with the same prefix name:
![image](https://user-images.githubusercontent.com/6134894/48998936-5d062c80-f155-11e8-805a-8c8d52a85ed7.png)

But I want to create an alert by the average of this two instances, so I've set an aggregation:
![image](https://user-images.githubusercontent.com/6134894/48999011-963e9c80-f155-11e8-8da9-29fee52ef2cb.png)

The problem is that when the alert is checked the aggregation is ignored and the alert is checked by for each instance instead of check the alert by the mean of the instances:
![image](https://user-images.githubusercontent.com/6134894/48999119-f6354300-f155-11e8-9482-0119dcfdbcd6.png)

I'm using Grafana 5.3.2 too.

Should I create a new issue?

**Comment by sunker** [comment]
Yes @bsmarcosj, please create a new issue. 

@amit-o sounds like this issue can be closed then?

**Comment by amit-o** [comment]
@sunker Well, this problem is still present on my end and 100% reproducible.
Since you cannot reproduce it, I'm not sure what the next course of action should be.

**Comment by daniellee** [comment]
Fixed by #14735
